extendr-api/src/ownership.rs

```diff
@@ -267,6 +267,16 @@ impl Ownership {
     }
 }
 
+impl Ownership {
+    /// Return the total number of currently protected elements, i.e. those with
+    /// non-zero reference count.
+    pub fn total_protected(&self) -> usize {
+        self.objects
+            .values()
+            .fold(0, |acc, x| if x.refcount != 0 { acc } else { acc + 1 })
```

Comment:
I think this is wrong, you count items with zero references, no? Also you can just do filter + count, much more readable.

Reply:
Nice you're looking at this! I need all the feedback I can receive for this task...

I wrote this to have something in the test... But I realize now, that I'm not actually using this in the test, because the 
test does not use the `ownership` module. So right now, this is actually very redundant.

Next, what's the logic here? I want to count the total number of protected items. A protected item, is an item, who has a reference count that is non-zero, because that means, it is being actively protected. If the reference count is non-zero, then I need to count it, and if the reference count is exactly zero, then I should disregard it.

Well, that is the opposite of what I wrote up there. So thank you for the tip!